processor/deltatocumulativeprocessor/internal/data/expo_test.go

```diff
@@ -85,9 +85,33 @@ func TestExpoAdd(t *testing.T) {
 			bs := pmetric.NewExponentialHistogramDataPointBuckets()
 			expotest.ObserveInto(bs, expo.Scale(0), 1, 2, 3, 4)
 			expotest.ObserveInto(bs, expo.Scale(0), 4, 3, 2, 1)
-			bs.BucketCounts().Append([]uint64{0, 0}...) // rescaling leaves zeroed memory. this is expected
 			return bs
 		}()},
+	}, {
+		name: "scale/no_downscale_within_limit",
+		dp:   expdp{Scale: 0, PosNeg: generateBins(10, 80, 1), Count: 160},
+		in:   expdp{Scale: 0, PosNeg: generateBins(80+10, 60, 2), Count: 120},
+		want: expdp{Scale: 0, PosNeg: generateBins(10, 80, 1, 60, 2), Count: 280},
```

Comment:
why do you need `generateBins` and `downscaled`? 
From looking at this test case, it's hard to reason about it. I cannot clearly see what either the input or the output looks like and as such not tell whether it's correct.
Is there a way to stick to written-out `bins{}` literals?

Reply:
made `maxBuckets` configurable in code (var instead of const), updated the test to set it to max 8 buckets and moved to `rawbs` method
should be easier to read now, please let me know if it needs any further changes!